src/charm.py

```diff
@@ -8,18 +8,136 @@
+                    "summary": "Oathkeeper Operator layer",
+                    "command": f"oathkeeper serve -c {self._oathkeeper_config_path}",
+                    "startup": "enabled",
+                }
+            },
+            "checks": {
+                "alive": {
+                    "override": "replace",
+                    "http": {"url": f"http://localhost:{OATHKEEPER_API_PORT}/health/alive"},
+                },
+                "ready": {
+                    "override": "replace",
+                    "http": {"url": f"http://localhost:{OATHKEEPER_API_PORT}/health/ready"},
+                },
+            },
+        }
+        return Layer(layer_config)
+
+    @property
+    def _oathkeeper_service_is_created(self) -> bool:
+        try:
+            self._container.get_service(self._container_name)
+        except (ModelError, RuntimeError):
+            return False
+        return True
+
+    def _render_access_rules_file(self) -> str:
+        """Render the access rules file."""
+        with open("templates/access-rules.yaml.j2", "r") as file:
+            template = Template(file.read())
+
+        rendered = template.render(
+            kratos_login_url=self._kratos_login_url,
+            return_to="http://default-url.com",
+        )
+        return rendered
+
+    def _render_conf_file(self) -> str:
+        """Render the Oathkeeper configuration file."""
+        with open("templates/oathkeeper.yaml.j2", "r") as file:
+            template = Template(file.read())
+
+        rendered = template.render(
+            kratos_session_url=self._kratos_session_url,
+            kratos_login_url=self._kratos_login_url,
+        )
+        return rendered
+
+    def _on_oathkeeper_pebble_ready(self, event: PebbleReadyEvent) -> None:
+        """Event Handler for pebble ready event."""
+        if not self._container.can_connect():
+            event.defer()
+            logger.info("Cannot connect to Oathkeeper container. Deferring the event.")
+            self.unit.status = WaitingStatus("Waiting to connect to Oathkeeper container")
+            return
+
+        self.unit.status = MaintenanceStatus("Configuring the container")
+
+        self._container.add_layer(self._container_name, self._oathkeeper_layer, combine=True)
+
+        if not self._oathkeeper_service_is_created:
+            event.defer()
+            self.unit.status = WaitingStatus("Waiting for Oathkeeper service to be created")
+            logger.info("Oathkeeper service is absent. Deferring the event.")
+            return
+
+        self._container.push(
+            self._oathkeeper_access_rules_path, self._render_access_rules_file(), make_dirs=True
+        )
+        self._container.push(
+            self._oathkeeper_config_path, self._render_conf_file(), make_dirs=True
+        )
+
+        try:
+            self._container.start(self._container_name)
+        except ChangeError as err:
+            logger.error(str(err))
```

Comment:
```suggestion
            logger.error(err)
```

do we need the casting?

Reply:
Yes, I can't find the related issue but the errors produced by juju need to be casted to string.